extensions/metadata_exchange/plugin.h

```diff
@@ -63,11 +64,16 @@ class PluginRootContext : public RootContext {
 
   StringView metadataValue() { return metadata_value_; };
   StringView nodeId() { return node_id_; };
+  bool updatePeer(StringView key, StringView peer_id, StringView peer_header);
 
  private:
   void updateMetadataValue();
   std::string metadata_value_;
   std::string node_id_;
+
+  // maps peer ID to the decoded peer flat buffer
+  std::unordered_map<std::string, std::string> cache_;
```

Comment:
should this be an ordered map in terms of when the entry was added, so that when cache is resized it can be FIFO?

Reply:
Yeah, I just copied what was there in node_info_cache. I agree that it should not randomly remove 1/4 elements. I can take that as a task for later.

Reply:
I split the root context for MX into outbound and inbound to at least get partitioned cache.